modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistry.java

```diff
@@ -203,48 +335,108 @@ public Object lookupObject(String key)
         {
             // FBE is a result of a broken config, propagate it (see MULE-3297 for more details)
             String message = String.format("Failed to lookup beans of type %s from the Spring registry", type);
-            throw new MuleRuntimeException(MessageFactory.createStaticMessage(message), fbex);
+            throw new MuleRuntimeException(createStaticMessage(message), fbex);
         }
         catch (Exception e)
         {
-            logger.debug(e);
+            if (logger.isDebugEnabled())
+            {
+                logger.debug(e);
+            }
             return Collections.emptyMap();
         }
     }
 
-
-    ////////////////////////////////////////////////////////////////////////////////////
-    // Registry is read-only
-    ////////////////////////////////////////////////////////////////////////////////////
-
-    @Override
-    public void registerObject(String key, Object value) throws RegistrationException
+    protected <T> Map<String, T> lookupEntriesForLifecycle(Class<T> type)
     {
-        throw new UnsupportedOperationException("Registry is read-only so objects cannot be registered or unregistered.");
+        return internalLookupByTypeWithoutAncestors(type, false, false);
     }
 
-    @Override
-    public void registerObject(String key, Object value, Object metadata) throws RegistrationException
+    protected Map<String, Object> getDepencies(String key)
     {
-        throw new UnsupportedOperationException("Registry is read-only so objects cannot be registered or unregistered.");
-    }
+        if (applicationContext instanceof ConfigurableApplicationContext)
```

Comment:
Condition applicationContext instanceof ConfigurableApplicationContext is used many times. That condition is used also to set the value of readOnly property. It would be good to replace this condition with that property (maybe renamed)